### PR TITLE
Enable adding TIs to TI groups.

### PR DESCRIPTION
**Pull request by ndmckinley**
### Description
This makes it so that admins can add TIs to TI groups, and includes a browser test.

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary
Screen Shot 2021-05-03 at 2 39 21 PM
Screen Shot 2021-05-03 at 2 39 28 PM

### CODE REVIEW

File: universal-application-tool-0.0.1/app/controllers/admin/TrustedIntermediaryManagementController.java [hunk 1]
```diff
@@ -12,21 +15,27 @@
...
   @Inject
   public TrustedIntermediaryManagementController(
       TrustedIntermediaryGroupListView listView,
+      EditTrustedIntermediaryGroupView editView,
       UserRepository userRepository,
       FormFactory formFactory) {
     this.listView = listView;
     this.userRepository = userRepository;
     this.formFactory = formFactory;
+    this.editView = editView;
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/app/controllers/admin/TrustedIntermediaryManagementController.java:38]: (blocking) these should have `checkNotNull`s.

File: universal-application-tool-0.0.1/app/models/Account.java [hunk 2]
```diff
@@ -58,4 +59,11 @@ public void setManagedByGroup(TrustedIntermediaryGroup group) {
   public Optional<TrustedIntermediaryGroup> getManagedByGroup() {
     return Optional.fromNullable(this.managedByGroup);
   }
+
+  public String getApplicantName() {
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/app/models/Account.java:63]: (blocking) An `Account` has many `Applicant`s. Which one is this returning? It looks like it is the most recent applicant? 

Can you add a comment describing which applicant's name this is, and also change the method name to be more specific?

File: universal-application-tool-0.0.1/app/repository/UserRepository.java [hunk 3]
```diff
@@ -84,18 +86,21 @@ public UserRepository(
                     .collect(ImmutableList.toImmutableList()));
   }
 
+  public Optional<Account> lookupAccount(String emailAddress) {
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/app/repository/UserRepository.java:89]: (blocking) should this be async?

File: universal-application-tool-0.0.1/app/repository/UserRepository.java [hunk 4]
```diff
@@ -166,4 +171,57 @@ public TrustedIntermediaryGroup createNewTrustedIntermediaryGroup(
     tiGroup.save();
     return tiGroup;
   }
+
+  public void deleteTrustedIntermediaryGroup(long id) throws NoSuchTrustedIntermediaryGroupError {
+    Optional<TrustedIntermediaryGroup> tiGroup =
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/app/repository/UserRepository.java:176]: This should use the `getTrustedIntermediaryGroup` method.

File: universal-application-tool-0.0.1/app/repository/UserRepository.java [hunk 5]
```diff
@@ -166,4 +171,57 @@ public TrustedIntermediaryGroup createNewTrustedIntermediaryGroup(
...
+    account.save();
+  }
+
+  public void removeTrustedIntermediaryFromGroup(long id, long accountId)
+      throws NoSuchTrustedIntermediaryGroupError, NoSuchTrustedIntermediaryError {
+    Optional<TrustedIntermediaryGroup> tiGroup = getTrustedIntermediaryGroup(id);
+    if (tiGroup.isEmpty()) {
+      throw new NoSuchTrustedIntermediaryGroupError();
+    }
+    Optional<Account> accountMaybe =
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/app/repository/UserRepository.java:213]: (non-blocking) there could be an overloaded `lookupAccount(long accountId)` method.

File: universal-application-tool-0.0.1/conf/routes [hunk 6]
```diff
@@ -49,8 +49,12 @@ GET     /admin/questions/:id/translations/:locale/edit  controllers.admin.AdminQ
 POST    /admin/questions/:id/translations/:locale       controllers.admin.AdminQuestionTranslationsController.update(request: Request, id: Long, locale: String)
 
 # Trusted Intermediary routes
-GET     /admin/tis               controllers.admin.TrustedIntermediaryManagementController.index(request: Request)
-POST    /admin/tis/new           controllers.admin.TrustedIntermediaryManagementController.create(request: Request)
+GET     /admin/tis                    controllers.admin.TrustedIntermediaryManagementController.index(request: Request)
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/conf/routes:52]: (blocking) the resource here is a `trusted intermediary group` right? I think `tis` is not descriptive enough. It should at least be `tiGroup`, if not the whole thing.

File: universal-application-tool-0.0.1/conf/routes [hunk 7]
```diff
@@ -49,8 +49,12 @@ GET     /admin/questions/:id/translations/:locale/edit  controllers.admin.AdminQ
...
+POST    /admin/tis/new                controllers.admin.TrustedIntermediaryManagementController.create(request: Request)
+POST    /admin/tis/:id/delete         controllers.admin.TrustedIntermediaryManagementController.delete(id: Long, request: Request)
+GET     /admin/tis/:id/edit           controllers.admin.TrustedIntermediaryManagementController.edit(id: Long, request: Request)
+POST    /admin/tis/:id/addTi          controllers.admin.TrustedIntermediaryManagementController.addIntermediary(id: Long, request: Request)
+POST    /admin/tis/:id/removeTi/:tiId controllers.admin.TrustedIntermediaryManagementController.removeIntermediary(id: Long, tiId: Long)
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/conf/routes:57]: (blocking): I think this should be submitting a hidden form with a `tiId` field set.

**Review by kendrickt — CHANGES_REQUESTED** [review]
Can you add screenshots?

File: universal-application-tool-0.0.1/conf/routes (hunk 7 above)
Comment (ndmckinley) [reply universal-application-tool-0.0.1/conf/routes]: Sure, added that to the button class.

File: universal-application-tool-0.0.1/app/repository/UserRepository.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (ndmckinley) [reply universal-application-tool-0.0.1/app/repository/UserRepository.java]: Done.

File: universal-application-tool-0.0.1/app/repository/UserRepository.java [hunk 8]
```diff
@@ -166,4 +171,57 @@ public TrustedIntermediaryGroup createNewTrustedIntermediaryGroup(
...
+      throw new NoSuchTrustedIntermediaryGroupError();
+    }
+    ebeanServer.delete(tiGroup.get());
+  }
+
+  public Optional<TrustedIntermediaryGroup> getTrustedIntermediaryGroup(long id) {
+    return ebeanServer.find(TrustedIntermediaryGroup.class).setId(id).findOneOrEmpty();
+  }
+
+  public void addTrustedIntermediaryToGroup(long id, String emailAddress)
```
[the remark this answers is not in the document]
Comment (ndmckinley) [reply universal-application-tool-0.0.1/app/repository/UserRepository.java:192]: Sure, done.

File: universal-application-tool-0.0.1/app/models/Account.java (hunk 2 above)
Comment (kendrickt) [reply universal-application-tool-0.0.1/app/models/Account.java:69]: ahh that makes sense. i was confused about why we max on time.

**Review by kendrickt — APPROVED** [review]

**Comment by ndmckinley** [comment]
Added screenshots - merging.

**Pull request merged**